test/responses_unit_test.go

```diff
@@ -0,0 +1,223 @@
+		case "lookupAssetBalances":
+			response, err = indexerC.LookupAssetBalances(10).Do(context.Background())
+		case "lookupAssetByID":
+			round, something, err = indexerC.LookupAssetByID(10).Do(context.Background())
+			response = models.LookupAssetByIDResponse{
+				CurrentRound: round,
+				Asset:        something.(models.Asset),
+			}
+		case "searchForAssets":
+			response, err = indexerC.SearchForAssets().Do(context.Background())
+		case "lookupAccountTransactions":
+			response, err = indexerC.LookupAccountTransactions("").Do(context.Background())
+		case "lookupAssetTransactions":
+			response, err = indexerC.LookupAssetTransactions(10).Do(context.Background())
+		case "searchForTransactions":
+			response, err = indexerC.SearchForTransactions().Do(context.Background())
+		case "any":
+			// This is an error case
+			// pickup the error as the response
+			_, response = indexerC.SearchForTransactions().Do(context.Background())
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	case "algod":
+		switch endpoint {
+		case "GetStatus":
+			response, err = algodC.Status().Do(context.Background())
+		case "WaitForBlock":
+			response, err = algodC.StatusAfterBlock(10).Do(context.Background())
+		case "TealCompile":
+			response, err = algodC.TealCompile([]byte{}).Do(context.Background())
+		case "RawTransaction":
+			var returnedTxid string
+			returnedTxid, err = algodC.SendRawTransaction([]byte{}).Do(context.Background())
+			response = txidresponse{TxID: returnedTxid}
+		case "GetSupply":
+			response, err = algodC.Supply().Do(context.Background())
+		case "TransactionParams":
+			var sParams types.SuggestedParams
+			sParams, err = algodC.SuggestedParams().Do(context.Background())
+			response = models.TransactionParams{
+				ConsensusVersion: sParams.ConsensusVersion,
+				Fee:              uint64(sParams.Fee),
+				GenesisID:        sParams.GenesisID,
+				Genesishash:      sParams.GenesisHash,
+				LastRound:        uint64(sParams.FirstRoundValid),
+				MinFee:           81560,
+			}
+		case "GetApplicationByID":
+			response, err = algodC.GetApplicationByID(10).Do(context.Background())
+		case "GetAssetByID":
+			response, err = algodC.GetAssetByID(10).Do(context.Background())
+		case "any":
+			// This is an error case
+			// pickup the error as the response
+			_, response = indexerC.SearchForTransactions().Do(context.Background())
+		default:
+			err = fmt.Errorf("unknown endpoint: %s", endpoint)
+		}
+	}
+	return err
+}
+
+type txidresponse struct {
+	TxID string `json:"txId"`
+}
+
+func theParsedResponseShouldEqualTheMockResponse() error {
+	var err error
+	var responseJson string
+
+	if expectedStatus != 200 {
+		responseJson = response.(error).Error()
+		// The error message is not a well formed Json.
+		// Verify the expected status code, and remove the json corrupting message
+		extraTxt := fmt.Sprintf("HTTP %d Internal Server Error:", expectedStatus)
```

Comment:
Could you implement this in a way that supports other error codes?

For example `responseJson = strings.Split(responseJson, ':')[1]`
or `responseJson = responseJson[strings.Index(responseJson, "{"):]`

Reply:
done